core/dbt/cli/flags.py

```diff
@@ -289,6 +289,15 @@ def _assign_params(
             params_assigned_from_default, ["WARN_ERROR", "WARN_ERROR_OPTIONS"]
         )
 
+        # Handle arguments mutually exclusive with INLINE
+        if hasattr(self, "INLINE"):
+            if hasattr(self, "SELECT"):
```

Comment:
hmm, presumably we didn't need to do this handling for WARN_ERROR or WARN_ERROR_OPTIONS because they are global flags. 

What do you think about baking it into `_assert_mutually_exclusive`? 

e.g. 
```python
def _assert_mutually_exclusive(
        self, params_assigned_from_default: Set[str], group: List[str]
    ) -> None:
        """
        Ensure no elements from group are simultaneously provided by a user, as inferred from params_assigned_from_default.
        Raises click.UsageError if any two elements from group are simultaneously provided by a user.
        """
        set_flag = None
        for flag in group:
            flag_set_by_user = hasattr(self, flag.lower()) and flag.lower() not in params_assigned_from_default
...
```

and simplifying these lines to: 
```python
self._assert_mutually_exclusive(params_assigned_from_default, ["SELECT", "INLINE"])
self._assert_mutually_exclusive(params_assigned_from_default, ["SELECTOR", "INLINE"])
```
(I don't think the `self.SELECTOR is not None` is necessary since default handling already occurs in _assert_mutually_exclusive)

Reply:
Yes, that's better. Have changed.